py/objproperty.c

```diff
@@ -38,28 +38,24 @@ typedef struct _mp_obj_property_t {
 } mp_obj_property_t;
 
 STATIC mp_obj_t property_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 4, false);
+    enum { ARG_fget, ARG_fset, ARG_fdel, ARG_doc };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+        { MP_QSTR_doc, MP_ARG_OBJ, {.u_rom_obj = mp_const_none} },
+    };
+    mp_arg_val_t vals[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all_kw_array(n_args, n_kw, args, MP_ARRAY_SIZE(allowed_args), allowed_args, vals);
 
     mp_obj_property_t *o = m_new_obj(mp_obj_property_t);
     o->base.type = type;
-    if (n_args >= 4) {
+    if (vals[ARG_doc].u_obj != mp_const_none) {
```

Comment:
Remove this line (and empty {} block) and simply add a comment after the assignment of ARG_fdel that the doc arg is ignored.  It's simpler and still obvious what's going on.